src/adagrams.js

```diff
@@ -1,15 +1,154 @@
 export const drawLetters = () => {
-  // Implement this method for wave 1
+let lettersDictionary = {
+    "A": 9, 
+    "B": 2, 
+    "C": 2, 
+    "D": 4, 
+    "E": 12, 
+    "F": 2, 
+    "G": 3, 
+    "H": 2, 
+    "I": 9, 
+    "J": 1, 
+    "K": 1, 
+    "L": 4, 
+    "M": 2, 
+    "N": 6, 
+    "O": 8, 
+    "P": 2, 
+    "Q": 1, 
+    "R": 6, 
+    "S": 4, 
+    "T": 6, 
+    "U": 4, 
+    "V": 2, 
+    "W": 2, 
+    "X": 1, 
+    "Y": 2, 
+    "Z": 1
+    };
+let count = 0;
+const characters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+let lettersInHand = [];
+while (count < 10){
+  let letterChosen = characters.charAt(Math.floor(Math.random() * 26));
+  if (lettersDictionary[letterChosen] > 0){
+    lettersInHand.push(letterChosen);
+    count += 1;
+    lettersDictionary[letterChosen] -= 1;
+  }
+}
+return lettersInHand;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+//Code to create the dictionary of the letters in hand
+  const dictOfLettersInHand = {}
+  for (let i=0; i< lettersInHand.length ; i++){
+    if (lettersInHand[i] in dictOfLettersInHand){
+
+      let value = dictOfLettersInHand[lettersInHand[i]] ;
+      dictOfLettersInHand[lettersInHand[i]] = value +1;
+    }
+    else {
+      dictOfLettersInHand[lettersInHand[i]] = 1;
+    }
+  }
+  //code to check occuerences and no. of occurences of characters 
+  //in input in the dictionary 
+  for (let i=0; i< input.length; i++){
+  if(dictOfLettersInHand[input[i]] == null){
+    return false;
+  } else if (dictOfLettersInHand[input[i]]===0)
+  {
+    return false;
+  }
+  dictOfLettersInHand[input[i]] -= 1;
+  }
+  return true; 
 };
 
 export const scoreWord = (word) => {
-  // Implement this method for wave 3
+  let lettersDictionary = {
+    "A": 1, 
+    "B": 3, 
+    "C": 3, 
+    "D": 2, 
+    "E": 1, 
+    "F": 4, 
+    "G": 2, 
+    "H": 4, 
+    "I": 1, 
+    "J": 8, 
+    "K": 5, 
+    "L": 1, 
+    "M": 3, 
+    "N": 1, 
+    "O": 1, 
+    "P": 3, 
+    "Q": 10, 
+    "R": 1, 
+    "S": 1, 
+    "T": 1, 
+    "U": 1, 
+    "V": 4, 
+    "W": 4, 
+    "X": 8, 
+    "Y": 4, 
+    "Z": 10
+    };
+    let pointsForWord = 0;
+    let upperCaseWord = word.toUpperCase();
```

Comment:
I suggest placing these variable declarations under the if statement below so they are only created if they will be used.